apps/andi/lib/andi_web/live/ingestion_live_view/transformations/transformation_form.ex

```diff
@@ -51,32 +52,25 @@ defmodule AndiWeb.IngestionLiveView.Transformations.TransformationForm do
 
           <div class="transformation-form__type">
             <%= label(f, :type, DisplayNames.get(:transformationType), class: "label label--required") %>
-            <%= select(f, :type, get_transformation_types(), phx_click: "transformation-type-selected", class: "select") %>
+            <%= select(f, :type, get_transformation_types(), phx_click: "transformation-type-selected", class: "select transformation-type") %>
             <%= ErrorHelpers.error_tag(f.source, :type, bind_to_input: false) %>
           </div>
           <div class="transformation-form__fields">
             <%= for field <- get_fields(@transformation_type) do %>
               <%= TransformationFieldBuilder.build_input(field, assigns, f) %>
-            <%= end %>
+            <% end %>
           </div>
         </div>
     </form>
     """
   end
 
-  def handle_event("move-transformation", %{"id" => transformation_id, "move-index" => move_index}, socket) do
-    AndiWeb.Endpoint.broadcast_from(self(), "move-transformation", "move-transformation", %{
-      "id" => transformation_id,
-      "move-index" => move_index
-    })
-
-    {:noreply, socket}
-  end
-
```

Comment:
I'm having trouble understanding why `move-transformation` was removed here. Is it because now a move makes a change to the changeset, and when the updated changeset is re-rendered, the order of transformations is updated as part of that? So it no longer needs an explicit event handler?

Even though the handle event for `move-transformation` is removed here, it looks like `move-transformation` is still referenced in the phx-click on like 40. Does that need to be removed as well?

Reply:
Wow okay kuddos for you for picking this up as one of your first cards this event stuff is a lot.

The `move-transformation` handler was removed here because it is now present in the `transformations_step` component, indicated by the form element that sends the event targeting `#transformations-form`

`#transformations-form` being the top level ID in the `transformations_step` component threw me off a bit. I think unifying those names would require an unfortunate amount of changes so, I'll make a refactor card to align those and that might help readability.